shell/gpu/gpu_surface_metal.h

```diff
@@ -37,6 +36,10 @@ class SK_API_AVAILABLE_CA_METAL_LAYER GPUSurfaceMetal : public Surface {
   // external view embedder is present.
   bool render_to_surface_;
 
+  // Accumulated damage for each framebuffer; Key is address of underlying
+  // MTLTexture for each drawable
+  std::map<uintptr_t, SkIRect> damage_;
```

Comment:
If you were to create a class for this say `MetalFrameDamage`, which contains this map and has a couple of methods:

* `typedef uintptr_t MetalTextureId`
* `AccumulateDamage(MetalTextureId texture_to_exclude)`
* `ResetDamage(MetalTextureId texture)`
* `std::optional<SkRect> GetDamage(MetalTextureId texture)`

This would help us unit test these methods better and make it more readable IMO.